bin/cheribsdtest/cheribsdtest_fptr_canon.c

```diff
@@ -82,3 +84,21 @@ CHERIBSDTEST(fptr_canon_dlfunc,
 
 	cheribsdtest_success();
 }
+
+CHERIBSDTEST(fptr_canon_int,
+    "Check that function pointers are canonical when relocated as integer "
+    "addresses")
+{
+	volatile ptraddr_t fptr_int;
+	void (* volatile fptr_dlsym)(void);
+
+	fptr_int = cheribsdtest_dynamic_get_dummy_fptr_addr();
+	fptr_dlsym = (void (*)(void))dlsym(RTLD_DEFAULT,
+	    "cheribsdtest_dynamic_dummy_func");
+
+	CHERIBSDTEST_VERIFY2(fptr_int == (ptraddr_t)fptr_dlsym,
+	    "Integer address %#p differs from dlsym %#p",
```

Comment:
Hmm, I would just use `%p` for `fptr_int` here.  We know it won't have any attributes?  Alternatively, perhaps print it as a hex uintmax_t instead of a pointer?